### ISSUE TITLE
Incorrect type of string with example in schema

### ISSUE THREAD

**Issue by ApaDoctor** [RonnyPfannschmidt/prance#33]
In YAML you can write strings without quotes. For some specific cases you are forced to use quotes, for example when you put digits, but you want them to be of type string.
But when you write schema and set type of variable to string - when you provide `example` you expect that it will also be of type string. But it's converted to the python as integer.
## Expected Behaviour
Some schema in `requestBody` for example:
```yml
  type: object
      properties:
          some_variable:
             type: string
             example: 1111111111 # This should be string
      required: true
```
This should be:
```python
{'some_variable':'1111111111'}
```

## Actual Behaviour
```yml
  type: object
      properties:
          some_variable:
             type: string
             example: 1111111111
      required: true
```
Result:
```python
{'some_variable': 1111111111}
```
@jfinkhaeuser

**Comment by jfinkhaeuser** [comment]
That's not a bug, that's how the different specs work together.

YAML has no concept of the schemas you provide, it just has concepts of different data types. A sequence of digits is an [Integer](https://yaml.org/type/int.html), as described in the specs (base 10 is detected).

It's OpenAPI that assigns meaning to the different fields. In the [Schema Object](https://swagger.io/specification/#schemaObject) (current version of the specs), the `example` field can have any type, and is described as "freeform", and can contain strings for example when YAML or JSON can't be used to describe the example accurately. That is, the field value is not validated.

The TL;DR is, that neither YAML nor OpenAPI cares about the field enough to try and convert it. Unfortunately, that's not a bug, then, but working according to specification.

Now... I do agree that it'd be nice to have this feature! But it can get arbitrarily complex to implement. I'll mark this as an enhancement, but unless there's sufficient interest, I'll treat it as low priority. Sorry!

**Comment by RonnyPfannschmidt** [comment]
closing as spec provided footgun
